python_modules/dagster/dagster/check/__init__.py

```diff
@@ -226,6 +226,12 @@ def opt_str_param(obj, param_name, default=None):
     return default if obj is None else obj
 
 
+def opt_nonempty_str_param(obj, param_name, default=None):
```

Comment:
what does this do that str_param does not?

Reply:
if we pass `''`, this returns `default` instead of `''`